packages/office-ui-fabric-react/src/components/Button/_base/Button.base.tsx

```diff
@@ -0,0 +1,265 @@
+} from '../../../Utilities';
+
+const getClassNames = classNamesFunction<IButtonBaseStyleProps, IButtonBaseStyles>();
+
+@customizable('ButtonBase', ['theme'])
+export class ButtonBase extends BaseComponent<IButtonBaseProps, {}> implements IButtonBase {
+
+  public static defaultProps = {
+  };
+
+  private _classNames: {[key in keyof IButtonBaseStyles]: string };
+  private _buttonElement: HTMLElement;
+  private _labelId: string;
+  private _descriptionId: string;
+  private _ariaDescriptionId: string;
+
+  constructor(props: IButtonBaseProps, rootClassName: string) {
+    super(props);
+
+    this._labelId = getId();
+    this._descriptionId = getId();
+    this._ariaDescriptionId = getId();
+  }
+
+  public render(): JSX.Element {
+    const {
+      theme,
+      className,
+      disabled,
+      href,
+      checked,
+      primary,
+      getStyles,
+      expanded,
+      onRenderIcon = this._onRenderIcon,
+      onRenderAriaDescription = this._onRenderAriaDescription,
+      onRenderChildren = this._onRenderChildren,
+      onRenderMenuIcon = this._onRenderMenuIcon,
+      onRenderPrefix,
+      onRenderSuffix
+    } = this.props;
+
+    // Anchor tag cannot be disabled hence in disabled state rendering
+    // anchor button as normal button
+    const renderAsAnchor: boolean = !disabled && !!href;
+
+    const nativeProps = this._getNativeProps(this.props, renderAsAnchor);
+
+    const {
+      ariaDescribedBy,
+      ariaLabelledBy,
+      ariaLabel
+    } = this._getAriaProps(
+        this.props,
+        getNativeProps(this.props, [renderAsAnchor ? 'a' : 'button']
+        ));
+
+    this._classNames = getClassNames(getStyles!, { theme: theme!, className, disabled, checked, expanded, primary });
+
+    let RootType = renderAsAnchor ? 'a' : 'button';
+    return (
+      <div className={ this._classNames.root }>
+        { onRenderPrefix && onRenderPrefix() }
+
+        <RootType
+          {...nativeProps}
+          className={ this._classNames.button }
+          type={ !renderAsAnchor && 'button' }
+          disabled={ disabled }
+          data-is-focusable={ ((this.props as any)['data-is-focusable'] === false || disabled) ? false : true }
+          aria-pressed={ checked }
+          aria-label={ ariaLabel }
+          aria-labelledby={ ariaLabelledBy }
+          aria-describedby={ ariaDescribedBy }
+        >
+          { onRenderIcon(this.props, this._onRenderIcon) }
+          { this._onRenderTextContents(this.props) }
+          { onRenderAriaDescription(this.props, this._onRenderAriaDescription) }
+          { onRenderChildren(this.props, this._onRenderChildren) }
+          { onRenderMenuIcon(this.props, this._onRenderMenuIcon) }
+        </RootType>
+
+        { onRenderSuffix && onRenderSuffix() }
+      </div>
+    );
+  }
+
+  public focus(): void {
+    if (this._buttonElement) {
+      this._buttonElement.focus();
+    }
+  }
+
+  public dismissMenu(): void {
+    this.setState({ menuProps: null });
+  }
```

Comment:
Bad merge? No menuProps anymore right?

Reply:
hah, yeah. I dont' even have a state to modify